bibliopixel/remote/control.py

```diff
@@ -0,0 +1,140 @@
+import multiprocessing
+import inspect
+from .. animation import BaseAnimation
+from .. animation.animation import COMPLETE_REASON
+from . import server
+from .. import log
+
+
+DEFAULT_OFF = 'OFF_ANIM'
+DEFAULT_ANIM_CONFIG = {
+    'bgcolor': '#00ff00',
+    'font_color': '#ffffff',
+    'display': None
+}
+
+
+class RemoteControl():
+    def __init__(self, base_config, animations):
+        # These start with the defaults
+        self.config = {
+            'title': 'BiblioPixel Remote',
+            'bgcolor': '#000000',
+            'font_color': '#ffffff',
+            'external_access': False,
+            'port': 5000
+        }
+
+        self.config.update(base_config)
+
+        self.ui_config = {
+            'bgcolor': self.config['bgcolor'],
+            'title': self.config['title']
+        }
+
+        self.q_send = multiprocessing.Queue()
+        self.q_recv = multiprocessing.Queue()
+
+        server_args = (
+            self.config['external_access'],
+            self.config['port'],
+            self.q_send,
+            self.q_recv
+        )
+
+        self.server = multiprocessing.Process(target=server.run_server,
+                                              args=server_args)
+
+        if not isinstance(animations, list):
+            raise ValueError('animations must be a list!')
+
+        self.current_animation_name = None
+        self.current_animation_obj = None
+        self.animations = animations
+        self.animation_objs = {}
+        anim_list = []
+        for anim in self.animations:
+            anim_cfg = dict(DEFAULT_ANIM_CONFIG)
+            anim_cfg.update(anim)
+            if not anim_cfg['display']:
+                anim_cfg['display'] = anim['name']
+            anim_cfg['name'] = ''.join(e for e in anim['name'] if e.isalnum())
+            anim_cfg['animation'].complete_callback = self.complete_callback
+            self.animation_objs[anim_cfg['name']] = anim_cfg['animation']
+            del anim_cfg['animation']  # no longer need here
```

Comment:
It means I'm deleting it from anim_cfg because the animation objects (BaseAnimation) have been saved to self.animation_objs. self.animations is the names a UI element config. self.animation_objs is a dict of the actual BaseAnimation type objects that match those names.